client/gui/InterfaceObjectConfigurable.cpp

```diff
@@ -373,14 +405,13 @@ std::shared_ptr<CIntObject> InterfaceObjectConfigurable::buildWidget(JsonNode co
 	{
 		return buildSlider(config);
 	}
-	if(type == "custom")
-	{
-		return const_cast<InterfaceObjectConfigurable*>(this)->buildCustomWidget(config);
-	}
-	return std::shared_ptr<CIntObject>(nullptr);
+
+	logGlobal->debug("Calling custom widget building function");
+	return const_cast<InterfaceObjectConfigurable*>(this)->buildCustomWidget(config);
 }
 
 std::shared_ptr<CIntObject> InterfaceObjectConfigurable::buildCustomWidget(const JsonNode & config)
 {
+	logGlobal->error("Default custom widget builder called");
```

Comment:
Probably better to write the reason why this is an error instead
```suggestion
	logGlobal->error("Attempting to create custom widget, but this window has no custom widgets defined!");
```